builtin/game/misc.lua

```diff
@@ -4,74 +4,42 @@
 -- Misc. API functions
 --
 
-local timers = {}
-local mintime
-local function update_timers(delay)
-	mintime = false
-	local sub = 0
-	for index = 1, #timers do
-		index = index - sub
-		local timer = timers[index]
-		timer.time = timer.time - delay
-		if timer.time <= 0 then
-			core.set_last_run_mod(timer.mod_origin)
-			timer.func(unpack(timer.args or {}))
-			table.remove(timers, index)
-			sub = sub + 1
-		elseif mintime then
-			mintime = math.min(mintime, timer.time)
-		else
-			mintime = timer.time
-		end
-	end
-end
+local _jobs = {}
+local _time = 0.0
+local _last = 0.0
 
-local timers_to_add
-local function add_timers()
-	for _, timer in ipairs(timers_to_add) do
-		table.insert(timers, timer)
+core.register_globalstep(function(dtime)
+	local _new = core.get_us_time() / 1000000
+	if _new > _last then
+		_time = _time + (_new - _last)
+	else
+		-- rollover
+		_time = _time + _new
 	end
-	timers_to_add = false
-end
+	_last = _new
 
-local delay = 0
-core.register_globalstep(function(dtime)
-	if not mintime then
-		-- abort if no timers are running
+	if #_jobs < 1 then
 		return
 	end
-	if timers_to_add then
-		add_timers()
-	end
-	delay = delay + dtime
-	if delay < mintime then
-		return
+
+	for i = #_jobs, 1, -1 do
```

Comment:
It seems like this is done in reverse order so that the `table.remove` call doesn't mess things up, but that's not immediately apparent.  Could you add a comment describing why it's in reverse order?